### PR TITLE
change event shape to include test info and response shape

### PR DESCRIPTION
**Pull request by scogra17**
Co-authored-by: Katarina Rosiak <katarinarosiak@gmail.com>
Co-authored-by: Miles Abbason <miles.abbason@gmail.com>
Co-authored-by: Tim Dronkers <tdronkers@gmail.com>

Changes: 
* Add `title` to the event param, as this is what's needed for the v0 design. In future designs we may need to include an id from postgres 
* Add `actualValue` to the result object
* Add `title` to the result object
* Simplify `TestConfiguration.checkAssertions()`
* Update the eslint config file

# Validation 
Ran locally using `npm run locally`, which returned this response: 
```
{
  title: 'My new test',
  results: [
    {
      assertionType: 'statusCode',
      targetValue: 200,
      actualValue: 200,
      comparisonType: 'equals',
      success: true
    },
    {
      assertionType: 'responseTime',
      targetValue: 600,
      actualValue: 417,
      comparisonType: 'less_than',
      success: true
    },
    {
      assertionType: 'headers',
      targetValue: [Array],
      actualValue: null,
      success: null,
      error: 'Unrecognized assertion type'
    },
    {
      assertionType: 'jsonBody',
      targetValue: [Array],
      actualValue: null,
      success: null,
      error: 'Unrecognized assertion type'
    }
  ]
}
```

### CODE REVIEW

File: entities/TestConfiguration.js [hunk 1]
```diff
@@ -1,72 +1,79 @@
...
+    this.assertions = httpRequest.assertions;
   }
 
   checkAssertions(response) {
     const results = [];
-    let result;
+    let targetValue;
+    let actualValue;
+    let comparisonType;
+    let success;
```
Comment (mabbason) [inline entities/TestConfiguration.js:17]: Can this be written `let targetValue, actualValue, comparisonType, success;` ? Idk, not suggesting per se. I thought of this, then wondered if maybe it was poor style

File: entities/TestConfiguration.js [hunk 2]
```diff
@@ -1,72 +1,79 @@
...
+            success: null,
             error: 'Unrecognized assertion type',
           });
       }
     }
     return results;
   }
 
-  static checkResTimeOrStatusCode(actual, comparison, target) {
+  static checkResTimeOrStatusCode({ targetValue, actualValue, comparisonType }) {
```
Comment (mabbason) [inline entities/TestConfiguration.js:62]: just curious here, why is this one a `static` method?

File: entities/TestConfiguration.js (hunk 1 above)
Comment (scogra17) [reply entities/TestConfiguration.js:17]: Was in the midst of changing it, then eslint growled...
Screen Shot 2022-07-12 at 1 43 23 PM

Comment (mabbason) [reply entities/TestConfiguration.js:17]: Perfect, that gut check was prolly from eslint yelling at me before

File: entities/TestRunner.js [hunk 3]
```diff
@@ -20,13 +20,9 @@ class TestRunner {
...
     const response = await this.httpRequest();
-    const result = this.testConfiguration.checkAssertions(response);
+    const results = this.testConfiguration.checkAssertions(response);
 
-    return {
-      statusCode: 200,
-      responseTime: response.headers['request-duration'],
-      body: result,
-    };
+    return results;
```
Comment (tdronkers04) [inline entities/TestRunner.js:25]: yes, this is definitely better.

**Pull request merged**